### PR TITLE
Improve variable references

### PR DESCRIPTION
**Pull request by tpdownes**
- Create HclString function as single implementation for converting variable references to HCL representation.
- create a const for string used in several related tests

### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [ ] Have you written unit tests to cover this change?
* [x] Is unit test coverage still above 80%?
* [x] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: pkg/config/expand.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ const (
 	blueprintLabel  string = "ghpc_blueprint"
 	deploymentLabel string = "ghpc_deployment"
 	roleLabel       string = "ghpc_role"
+	globalGroupID   string = "deployment"
```
Comment (mr0re1) [inline pkg/config/expand.go:35]: IIUC we never surface this "deployment" string to the user, at the same time it prevents (or even worse silently allows, but makes mistakes) users from naming they groups "deployment".
This change can be a good opportunity to solve this problem. As a simple solution can we replace this with less usable word e.g. "hpc_toolkit_internal_keyword_for_global_variables".

File: pkg/config/expand.go [hunk 2]
```diff
@@ -630,6 +638,17 @@ func (ref varReference) ToModuleID() string {
...
 
+func (ref varReference) HclString() string {
+	switch ref.toGroupID {
+	case globalGroupID:
+		return "var." + ref.name
+	case ref.fromGroupID:
+		return "module." + ref.toModuleID + "." + ref.name
+	default:
+		return "var." + ref.name + "_" + ref.toModuleID
+	}
```
Comment (nick-stroud) [inline pkg/config/expand.go:649]: I feel like we lost some context with this function. 1. being what these cases stand for (global, within group, inter group) and 2. why the hcl is the way that it is. Part of me want to see the `IsIntergroup` function used but it does not lend itself to switch and the switch is also nice.

What about just adding some comments:
```
	case globalGroupID:
		return "var." + ref.name
	case ref.fromGroupID: 
		// intragroup reference can make direct reference to module output
		return "module." + ref.toModuleID + "." + ref.name
	default:  
		// for intergroup reference ghpc will create a variable
		return "var." + ref.name + "_" + ref.toModuleID
```

**Review by nick-stroud — APPROVED** [review]

File: pkg/config/expand.go (hunk 1 above)
Comment (tpdownes) [reply pkg/config/expand.go:35]: I think this has some value but I'd rather not (potentially) protract this particular PR. There is a current task to reject deployment groups that are named "deployment" and using a reserved keyword for this value may be a better way to address that.

File: pkg/config/expand.go (hunk 2 above)
Comment (tpdownes) [reply pkg/config/expand.go:652]: Addressed on HEAD of PR.

**Pull request merged**